rslib/src/config/mod.rs

```diff
@@ -286,6 +292,27 @@ pub(crate) enum Weekday {
     Saturday = 6,
 }
 
+#[derive(Debug, Clone, Copy, serde_derive::Deserialize, serde_derive::Serialize)]
+pub struct BackupLimits {
+    #[serde(rename = "d")]
+    pub daily: u32,
+    #[serde(rename = "w")]
+    pub weekly: u32,
+    #[serde(rename = "m")]
+    pub monthly: u32,
+}
+
+impl Default for BackupLimits {
+    fn default() -> Self {
+        // 2d + 12d + 10w + 9m ≈ 1y
```

Comment:
Feel free to tweak these to your liking.

Reply:
The way I'd been thinking it might work is that the ranges would overlap, rather than be one after another. That would allow a particular backup to satisfy multiple criteria at once (eg a given day's backup, but also a given month's backup), and wouldn't require adding the different periods up to know the maximum interval that will be kept. I'm not pushing for this solution over the existing one at the moment (haven't thought it through carefully yet :-), but wondered what you think.

Reply:
This approach simplifies calculating the number of stored backups, though.
Anyway, the main reason I chose it was the overlap of weeks and months: The backup we'd like to keep for January is the one from the 31st, which was a Monday. However, the backup we'd like to keep for that week is from Sunday, February 6. So we'd have to keep two backups from the same week.
I also thought that was your plan, too, because you wrote this:

> 5 might be a sensible default for x if we have a 48 hour exclusion zone, which should give ~1 week of daily backups.

Treating the grace period differently might increase confusion.

Reply:
That all sounds reasonable; let's run with your solution.